src/main/scala/is/hail/utils/BufferedAggregatorIterator.scala

```diff
@@ -15,7 +19,10 @@ class BufferedAggregatorIterator[T, U, K](
   private var remainder: Iterator[(K, U)] = Iterator.empty
 
   // TODO: use a heavy hitters buffer here
-  private val buffer = new util.LinkedHashMap[K, U](bufferSize, 0.75f, true) {
+  private val buffer = new util.LinkedHashMap[K, U](
+    (bufferSize / BufferedAggregatorIterator.loadFactor).toInt,
```

Comment:
I'm a little nervous that this is right on the edge, so that the buffer will be grown when the `bufferSize + 1`st element is added, which is the worst possible behavior. The completely safe thing to do would be to give the initial size `(bufferSize / BufferdAggregatorIterator.loadFactor).toInt` and load factor 1, so that it never grows and we keep it below the load factor ourselves.

Reply:
I can also just add one here